sklearn/discriminant_analysis.py

```diff
@@ -963,32 +1042,6 @@ def _decision_function(self, X):
         u = np.asarray([np.sum(np.log(s)) for s in self.scalings_])
         return -0.5 * (norm2 + u) + np.log(self.priors_)
 
-    def decision_function(self, X):
-        """Apply decision function to an array of samples.
-
-        The decision function is equal (up to a constant factor) to the
-        log-posterior of the model, i.e. `log p(y = k | x)`. In a binary
-        classification setting this instead corresponds to the difference
-        `log p(y = 1 | x) - log p(y = 0 | x)`. See :ref:`lda_qda_math`.
-
-        Parameters
-        ----------
-        X : array-like of shape (n_samples, n_features)
-            Array of samples (test vectors).
-
-        Returns
-        -------
-        C : ndarray of shape (n_samples,) or (n_samples, n_classes)
-            Decision function values related to each class, per sample.
-            In the two-class case, the shape is (n_samples,), giving the
-            log likelihood ratio of the positive class.
-        """
-        dec_func = self._decision_function(X)
-        # handle special case of two classes
-        if len(self.classes_) == 2:
-            return dec_func[:, 1] - dec_func[:, 0]
-        return dec_func
-
     def predict(self, X):
```

Comment:
We should need to define all those function below as well. It will use the one from the mixin